File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/20-Set-Map/LeetCode/LeetCode-0217-2.java
class Solution {
    // O(n) algorithm with O(n) extra space
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> mySet = new HashSet<>();

        for (var num: nums){
            if (mySet.contains(num)) // Expected O(1)
                return true;
            mySet.add(num); // Expected O(1)
        } // end-for

        return false;
    }
}

File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_323_Design_And_Analysis_Of_Algorithms/Algorithms/02-IterativeAlgorithms/LeetCode/LeetCode-0217-2.cpp
class Solution {
public:
    /// O(NlogN) algorithm with O(1) extra space
    bool containsDuplicate(vector<int>& nums) {
        std::sort(nums.begin(), nums.end());
        
        // Walk over the list & compare consecutive numbers
        for (int i=1; i<nums.size(); i++)
            if (nums[i] == nums[i-1])
                return true;

        return false;        
    } // end-containsDuplicate
};